### PR TITLE
Zero

### PR DESCRIPTION
**Pull request by gabeio**
A plugin to send zero extra energy, mainly great for moving silver around the universe.

### CODE REVIEW

**Comment by gabeio** [comment]
apparently not ready _just yet_.

**Comment by gabeio** [comment]
Ready 👍 sadly it isn't sending zero energy but only ~2. Apparently the landing function doesn't seem to be precise enough to use it to verify any energy below 2, it randomly works for some planets and others it will fail saying not enough forces even though landing says greater than 1 will be landing.

File: content/utilities/zero/plugin.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+class Plugin {
+  constructor() {
+	this.root = null;
+	this.container = null;
+  }
+
+  async render(container) {
+	this.container = container;
+	container.style.width = "215px";
+	this.root = render(html`<${App} />`, container);
```
Comment (phated) [inline content/utilities/zero/plugin.js:91]: don't need to assign to root anymore. You **do** need to clean up your preact component in destroy

File: content/utilities/zero/plugin.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  
+  useLayoutEffect(() => {
+    let onClick = () => {
+      setSelectedPlanet(ui.getSelectedPlanet());
+    }
+    window.addEventListener('click', onClick);
+
+    return () => {
+      window.removeEventListener('click', onClick);
+    }
```
Comment (phated) [inline content/utilities/zero/plugin.js:69]: This is now done with `ui.selectedPlanetId$.subscribe`:
```suggestion
    const sub = ui.selectedPlanetId$.subscribe(locationId => {
      setSelectedPlanet(df.getPlanetWithId(locationId));
    }

    return sub.unsubscribe;
```

**Comment by phated** [comment]
@gabeio Does this set up a sort of "Waypointing"? The current description has a lot of overlap with DistributeSilver and I think it would make a ton of sense to change this to "SilverWaypoint" and have it be a repeat-attack but for sending silver.

**Comment by gabeio** [comment]
@phated you were reading my mind, I was thinking the same thing (after using it for a while), that it would be really useful to make it repeat like repeat attack 😃

**Comment by jacobrosenthal** [comment]
:wave: @gabeio! you in round2? Do you wanna push this over the finish line?

**Comment by andrewhong5297** [comment]
Tested and this works for me!

File: content/utilities/zero/plugin.js (hunk 1 above)
Comment (jacobrosenthal) [inline content/utilities/zero/plugin.js:91]:
```suggestion
	render(html`<${App} />`, container);
```

File: content/utilities/zero/plugin.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+      <button onClick=${() => setDestination(getSelectedPlanet)}> Set Dest </button> <span>${planetShort(getDestination)}</span><br />
+	  <input type="text" style="color: black" value=${getSilver} onInput=${silverchange} placeholder="silver to send" /><button onClick=${() => setSilver("max")}> Max </button><br />
+	  <button onClick=${() => runZero(getSource, getDestination, getSilver)}> Send Zero </button>
+    </div>
+  `;
+}
+
+class Plugin {
+  constructor() {
+	this.root = null;
```
Comment (jacobrosenthal) [inline content/utilities/zero/plugin.js:84]:
```suggestion
```

File: content/utilities/zero/plugin.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+	this.container = null;
+  }
+
+  async render(container) {
+	this.container = container;
+	container.style.width = "215px";
+	this.root = render(html`<${App} />`, container);
+  }
+
+  destroy() {}
```
Comment (jacobrosenthal) [inline content/utilities/zero/plugin.js:94]: I think this is what @phated wanted
```suggestion
  destroy() {
      render(null, this.container);
  }
```

**Comment by jacobrosenthal** [comment]
Please reopen when you're back @gabeio !

**Pull request closed without merging**